### PR TITLE
[GCS] Fix the Windows build of GCS actor scheduling

### PR DESCRIPTION
**Pull request by Chong-Li**
## Why are these changes needed?

The GCS actor scheduling fails the Windows build of test_advanced. This PR fixes this error.

## Related issue number

#16580 #17941 #17944

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]
LGTM if it passes tests.

**Comment by ericl** [comment]
//python/ray/tests:test_advanced                                        TIMEOUT in 3 out of 3 in 300.3s
  Stats over 3 runs: max = 300.3s, min = 300.2s, avg = 300.2s, dev = 0.0s
  C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced/test.log
  C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced/test_attempts/attempt_1.log
  C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced/test_attempts/attempt_2.log
//python/ray/tests:test_advanced_client_mode                            TIMEOUT in 3 out of 3 in 300.3s
  Stats over 3 runs: max = 300.3s, min = 300.2s, avg = 300.2s, dev = 0.0s
  C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced_client_mode/test.log
  C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced_client_mode/test_attempts/attempt_1.log
  C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced_client_mode/test_attempts/attempt_2.log


Seems still failing.

**Comment by Chong-Li** [comment]
> //python/ray/tests:test_advanced TIMEOUT in 3 out of 3 in 300.3s
> Stats over 3 runs: max = 300.3s, min = 300.2s, avg = 300.2s, dev = 0.0s
> C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced/test.log
> C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced/test_attempts/attempt_1.log
> C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced/test_attempts/attempt_2.log
> //python/ray/tests:test_advanced_client_mode TIMEOUT in 3 out of 3 in 300.3s
> Stats over 3 runs: max = 300.3s, min = 300.2s, avg = 300.2s, dev = 0.0s
> C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced_client_mode/test.log
> C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced_client_mode/test_attempts/attempt_1.log
> C:/users/runneradmin/_bazel_runneradmin/vlncal46/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_advanced_client_mode/test_attempts/attempt_2.log
> 
> Seems still failing.

There are some scheduling-related test cases, such as `test_hybrid_policy`, skipped at "win32" because of timeout. What is the exact reason behind that "timeout". I was wondering whether it affects my e2e test cases. @ericl

**Comment by ericl** [comment]
@wuisawesome what was the original reason for skipping those?

Btw, does it work with higher timeouts?

**Comment by wuisawesome** [comment]
Hmmm, admittedly I don't remember any more. Perhaps we should investigate turning it back on for windows.

File: src/ray/common/task/scheduling_resources.cc [hunk 1]
```diff
@@ -190,8 +190,10 @@ const ResourceSet ResourceSet::GetNumCpus() const {
 }
 
 std::string format_resource(std::string resource_name, double quantity) {
-  if (resource_name == "object_store_memory" || resource_name == "memory") {
-    return std::to_string(quantity / (1024 * 1024 * 1024)) + " GiB";
+  if (resource_name == "object_store_memory" ||
+      resource_name.find(kMemory_ResourceLabel) == 0) {
+    // The memory resources (in 50MiB unit) are converted to GiB
+    return std::to_string(quantity * 50 / 1024) + " GiB";
```
Comment (scv119) [inline src/ray/common/task/scheduling_resources.cc:196]: why changing this?

**Pull request merged**